pkg/utils/shutdown.go

```diff
@@ -2,29 +2,113 @@ package utils
 
 import (
 	"context"
+	"time"
 
+	admv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
 )
 
-func Shutdown(c client.Client) {
-	shutdownLog := ctrl.Log.WithName("shutdown")
+var shutdownLog = ctrl.Log.WithName("shutdown")
+
+var sriovnetworksGVR = schema.GroupVersionResource{
+	Group:    "sriovnetwork.openshift.io",
+	Version:  "v1",
+	Resource: "sriovnetworks",
+}
+
+var failurePolicyIgnore = admv1.Ignore
+
+func Shutdown() {
+	updateFinalizers()
+	//updateWebhooks()
+}
+
+func getConfig() *rest.Config {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		shutdownLog.Error(err, "Error getting cluster config")
+	}
+	return config
+}
+
+func updateFinalizers() {
 	shutdownLog.Info("Clearing finalizers on exit")
-	networkList := &sriovnetworkv1.SriovNetworkList{}
-	err := c.List(context.TODO(), networkList)
+	c, err := dynamic.NewForConfig(getConfig())
```

Comment:
Can we use the `GetConfigOrDie` from "sigs.k8s.io/controller-runtime/pkg/client/config" instead `getConfig()`? So that we can support running the operator in both in-cluster and local mode. The local mode is quite handy for development.

Reply:
Done